packages/fuels/tests/scripts/script_generic_arguments/src/main.sw

```diff
@@ -0,0 +1,18 @@
+script;
+
+struct GenericBimbam<U> {
+    val: U,
+}
+
+struct GenericSnack<T,V> {
+    twix: GenericBimbam<T>,
+    mars: V,
+}
+
+fn main(bim: GenericBimbam<u8>, bam: GenericSnack<u16,u32>) -> GenericSnack<u64, u32> {
```

Comment:
It would be nice to see more complex types in the argument list. For example:
* `Vec`, `Option`, `Result`.
* Enum, array, string, tuple, `b256`, `bool`
I know all of these are tested for contracts, but we should make sure we don't regress for scripts as well.

Reply:
Yes I think it's important too, I will add these projects. `Vec` will not be supported yet because of the way it is encoded.

Reply:
Some of these test types I will add in a later PR